File: hydra-scala/src/main-gen/scala/hydra/evaluation/PrimitiveFunction.scala
package hydra.evaluation

import hydra.core.FunctionType
import hydra.core.Term

case class PrimitiveFunction(
    /**
     * @type function:
     *         from:
     *         - list: hydra/core.Term
     *         to: hydra/core.Term
     */
    implementation: Seq[Term] => Term,
    
    /**
     * @type hydra/core.FunctionType
     */
    `type`: FunctionType
)

File: hydra-java/src/main/java/hydra/PrimitiveFunction.java
package hydra;

import hydra.core.Name;
import hydra.core.Term;
import hydra.dsl.Terms;

public abstract class PrimitiveFunction<M> {
    public abstract Name name();

    public Term<M> term() {
        return Terms.primitive(name()); 
    }
}
